08_project/mohankrishna/pkg/mohankrishna-puppy/store/storer_test.go

```diff
@@ -0,0 +1,214 @@
+package store
+
+import (
+	"os"
+	"testing"
+
+	types "github.com/anz-bank/go-training/08_project/mohankrishna/pkg/mohankrishna-puppy"
+	tassert "github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storerImpl int
+
+const (
+	maps storerImpl = iota
+	smaps
+	levelDB
+)
+
+var (
+	puppy1 = func() types.Puppy {
+		return types.Puppy{
+			ID:     0x1234,
+			Breed:  "Sheep herder",
+			Colour: "Brown",
+			Value:  1000,
+		}
+	}
+	puppy2 = func() types.Puppy {
+		return types.Puppy{
+			ID:     0x1236,
+			Breed:  "Sheep herder",
+			Colour: "Brown",
+			Value:  1000,
+		}
+	}
+)
+
+type storerSuite struct {
+	suite.Suite
+	store Storer
+	impl  storerImpl
+}
+
+func (s *storerSuite) SetupSuite() {
+	switch s.impl {
+	case maps:
+		// create a map store
+		s.store = NewMapStore()
+	case smaps:
+		// create a sync map store
+		s.store = NewSyncStore()
+	case levelDB:
+		// create a level db store
+		s.store = NewLevelDBStore(os.TempDir() + "/level_store")
+	default:
+		panic("Unrecognised storer implementation")
+	}
+	puppy := puppy1()
+	err := s.store.CreatePuppy(&puppy)
+	if err != nil {
+		panic("Could not initialise tests")
+	}
+}
+
+func (s *storerSuite) TearDownSuite() {
+	os.RemoveAll(os.TempDir() + "/level_store")
+}
+
+func TestStorerImpls(t *testing.T) {
+	mapSuite := storerSuite{impl: maps}
+	suite.Run(t, &mapSuite)
+
+	syncMapSuite := storerSuite{impl: smaps}
+	suite.Run(t, &syncMapSuite)
+
+	levelDBSuite := storerSuite{impl: levelDB}
+	suite.Run(t, &levelDBSuite)
+}
+
+func (s *storerSuite) TestMapStoreReadPuppySuccessful() {
```

Comment:
Rename these functions to generic names since these are no longer `MapStore` specific.
eg: `TestStoreReadPuppySuccessful()`